spark/v3.2/spark/src/main/scala/org/apache/spark/sql/execution/datasources/SparkExpressionConverter.scala

```diff
@@ -30,4 +34,18 @@ object SparkExpressionConverter {
     // But these two conversions already exist and well tested. So, we are going with this approach.
     SparkFilters.convert(DataSourceStrategy.translateFilter(sparkExpression, supportNestedPredicatePushdown = true).get)
   }
+
+  @throws[AnalysisException]
+  def collectResolvedSparkExpression(session: SparkSession, tableName: String, where: String): Expression = {
```

Comment:
yeah it's hacky but I cannot think of any other better way, @RussellSpitzer any thoughts?

Reply:
I mean Ideally we could get all this into Spark so Spark could do these conversion for us and just give us filters, but I don't think there are any ways around this at the moment.

Reply:
Well one way I did do this previously, is I just made the full DatasourceRelationNode and just pulled the filters out of that. I'm not sure that's any cleaner than this

Reply:
yeah I think this would actually be cleaner than using DatasourceRelationNode, I am good with this impl, please let us know if you have any additional concern.

Reply:
Yep, lgtm. I think we may want to use similar logic in our expire snapshots procedure as well to allow functions and other Spark things

Reply:
ok, I will check expire snapshots procedure in separate PR. It is currently not the scope of this PR.